ecs-cli/modules/cli/configure/configure.go

```diff
@@ -29,6 +33,67 @@ func fieldEmpty(field string, flagName string) error {
 	return nil
 }
 
+// Migrate is the callback for the Migrate command
+func Migrate(context *cli.Context) error {
+	oldConfig := &config.CLIConfig{}
+	dest, err := config.NewDefaultDestination()
+	if err != nil {
+		return errors.Wrap(err, "Error reading old config file.")
+	}
+
+	// The iniReadWriter is rather terrible and will not throw parse errors
+	// See if we can read the config file as YAML, if that's true then we don't
+	// need to migrate
+	clusterConfig, err := config.ReadClusterFile(config.ConfigFilePath(dest))
+	if err == nil && clusterConfig.Version != "" {
+		// Is a new style config file
+		logrus.Fatalf("No need to migrate; found a YAML formatted config file at %s", config.ConfigFilePath(dest))
+	}
+
+	iniReadWriter, err := config.NewINIReadWriter(dest)
+	if err != nil {
+		return errors.Wrap(err, "Error reading old config file.")
+	}
+	err = iniReadWriter.GetConfig(oldConfig)
+	if err != nil {
+		return errors.Wrap(err, "Error reading old config file.")
+	}
+
+	if oldConfig.AWSProfile != "" {
+		logrus.Warnf("Storing AWS Profile in the config is no longer supported. Please use the %s flag inline in commands instead.", command.ProfileFlag)
+	}
+
+	if !context.Bool(command.ForceFlag) {
+		err = migrateWarning(oldConfig)
+		if err != nil {
+			return err
+		}
+		scanner := bufio.NewScanner(os.Stdin)
+		scanner.Scan()
+		input := scanner.Text()
+		if !strings.HasPrefix(input, "y") && !strings.HasPrefix(input, "Y") {
+			logrus.Info("Aborting Migration.")
+			return nil
+		}
+	}
+
+	cluster := &config.Cluster{Cluster: oldConfig.Cluster, Region: oldConfig.Region, CFNStackName: oldConfig.CFNStackName, ComposeServiceNamePrefix: oldConfig.ComposeServiceNamePrefix}
```

Comment:
We only want ComposeServiceNamePrefix if it is specified.

Reply:
If it is not specified in the old config, then it will be an empty string. Because Go initializes all string fields of structs to be empty strings, the code I wrote here should produce the same affect as if I had checked first if the prefix was empty. This way of doing it has the same affect, but its shorter code.